src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/HealthCertificateValidationService.swift

```diff
@@ -0,0 +1,555 @@
+			return completion(.failure(.TECHNICAL_VALIDATION_FAILED))
+		}
+		Log.info("Successfully passed technical validation. Proceed with updating value sets...", log: .vaccination)
+
+		proceedWithUpdatingValueSets(
+			healthCertificate: healthCertificate,
+			arrivalCountry: arrivalCountry,
+			validationClock: validationClock,
+			completion: completion
+		)
+	}
+	
+	// MARK: - Private
+	
+	private let store: Store
+	private let client: Client
+	private let vaccinationValueSetsProvider: VaccinationValueSetsProvider
+	private let signatureVerifier: SignatureVerification
+	private let validationRulesAccess: ValidationRulesAccessing
+	private var subscriptions = Set<AnyCancellable>()
+	
+	// MARK: - Flow
+	
+	private func proceedWithUpdatingValueSets(
+		healthCertificate: HealthCertificate,
+		arrivalCountry: String,
+		validationClock: Date,
+		completion: @escaping (Result<HealthCertificateValidationReport, HealthCertificateValidationError>) -> Void
+	) {
+		// 2. update/ download value sets
+		vaccinationValueSetsProvider.latestVaccinationCertificateValueSets()
+			.sink(
+				receiveCompletion: { result in
+					switch result {
+					case .finished:
+						break
+					case .failure(let error):
+						if case let URLSession.Response.Failure.httpError(_, response) = error {
+							switch response.statusCode {
+							case 500...509:
+								Log.error("Failed to update value sets with status code: \(response.statusCode)", log: .vaccination, error: error)
+								completion(.failure(.VALUE_SET_SERVER_ERROR))
+							default:
+								Log.error("Unhandled Status Code while fetching certificate value sets", log: .vaccination, error: error)
+								completion(.failure(.VALUE_SET_CLIENT_ERROR))
+							}
+							
+						} else if case URLSession.Response.Failure.noNetworkConnection = error {
+							Log.error("Failed to update value sets. No network error", log: .vaccination, error: error)
+							completion(.failure(.NO_NETWORK))
+						} else {
+							Log.error("Casting error for http error", log: .vaccination, error: error)
+							completion(.failure(.VALUE_SET_CLIENT_ERROR))
+						}
+					}
+					
+				}, receiveValue: { [weak self] valueSets in
+					Log.info("Successfully received value sets. Proceed with downloading acceptance rules...", log: .vaccination)
+					self?.proceedWithDownloadingAcceptanceRules(
+						healthCertificate: healthCertificate,
+						arrivalCountry: arrivalCountry,
+						validationClock: validationClock,
+						valueSets: valueSets,
+						completion: completion
+					)
+				}
+			)
+			.store(in: &subscriptions)
+	}
+	
+	private func proceedWithDownloadingAcceptanceRules(
+		healthCertificate: HealthCertificate,
+		arrivalCountry: String,
+		validationClock: Date,
+		valueSets: SAP_Internal_Dgc_ValueSets,
+		completion: @escaping (Result<HealthCertificateValidationReport, HealthCertificateValidationError>) -> Void
+	) {
+		// 3. update/ download acceptance rules
+		downloadRules(
```

Comment:
No, because we pass several properties (e.g. `arrivalCountry`) that are not needed for downloading the rules, but when the download is successful, the next method. If we would call downloadRules with these properties as parameters, i would be confused why the download needs all the properties.